hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -772,6 +776,7 @@ void finalizeExecution(@Nullable Throwable failure) {
                 ) {
                     mc.setJobStatus(SUSPENDED, description, userRequested);
                     mc.jobExecutionRecord().setSuspended(null);
+                    mc.coordinationService().unscheduleJobTimeout(mc.jobId());
```

Comment:
in previous implementaion, timeout could terminate also a suspended job. Now it does not, if I see correctly. We should not change this behavior.

Reply:
The original behavior doesn't make sense to me. We have the suspend-on-failure flag, which is used to re-run jobs by adding more resources or altering the configuration. Should the timeout continue during the maintenance? If this is the desired behavior, the suspend-on-failure shouldn't be used at all. Manually suspending a job implies even greater control and responsibility, in which we shouldn't interfere.

Reply:
I can see that you restored original behavior. For the record: reliable tracking of remaining time is hard, we would have to store it somewhere and update it regularly so it can survive node crash. This is bigger topic: job resource consumption monitoring and limiting, and it can be not only run time, but also CPU usage, memory usage, io usage etc. Maybe we will implement something like that in future version.